### PR TITLE
Strict boolean support for Strings in and out

### PR DESCRIPTION
**Pull request by bbakerman**
When a AST literal is coerced it must be `true` or `false` - easy because the parser enforces that

However if you use variables then

`{"someBoolVar" : "TEST"}`

will be accepted and evaluated to false.

The old code uses Boolean.parse in Java which I have argued IS idiomatic Java however it creates strangle edge cases

We could be more strict in what we accept as a boolean.

This is that PR

If we decide to go with this PR then it would break anyone how is currently sending in strings outside "true" and "false" (case insensitive) .

See 

https://github.com/graphql-java/graphql-java/issues/1103
https://github.com/graphql-java/graphql-java/issues/865

### CODE REVIEW

File: src/main/java/graphql/scalar/GraphqlBooleanCoercing.java [hunk 1]
```diff
@@ -22,7 +22,14 @@ private Boolean convertImpl(Object input) {
...
         } else if (input instanceof String) {
-            return Boolean.parseBoolean((String) input);
+            String lStr = ((String) input).toLowerCase();
+            if (lStr.equals("true")) {
+                return true;
+            }
+            if (lStr.equals("false")) {
+                return false;
+            }
+            return null;
```
Comment (bbakerman) [inline src/main/java/graphql/scalar/GraphqlBooleanCoercing.java:32]: null causes a coercing exception ins code you cant see in the diff

**Review by andimarek — APPROVED** [review]
